index.js

```diff
@@ -13,10 +16,15 @@ module.exports = function Rivet(url, options) {
     const _dependencies = processDependencies(dependencies)
     const _query = processVariables(dependencies, variables, query)
 
-    return client.request(
-      `${_query}\n${[..._fragments, ..._dependencies].join('\n')}`,
-      variables
-    )
+    const queryAsString = `${_query}\n${[..._fragments, ..._dependencies].join(
+      '\n'
+    )}`
+
+    if (retryCount) {
+      return requestWithRetry(retryCount, client, queryAsString, variables)
+    } else {
+      return client.request(queryAsString, variables)
+    }
   }
 
   fetch.client = client
```

Comment:
@jescalan when using Rivet like so `rivetQuery.client`, this client object won't have a retry mechanism. Right? Shouldn't it be covered by the retry mechanism too? 🤔

Reply:
fair point, i can update this!

Reply:
done here: https://github.com/hashicorp/rivet-graphql/commit/a75e45b5dc5ee4c62666dd33943f1dcc7750df54